### PR TITLE
 migration(palette v3): space 🛰

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Feature**

This PR resolves [CX-1705]

### Description

Enabling v3 for space globally, with the option to ask for v2.

v3
Screenshot 2021-08-02 at 10 40 27

v2
Screenshot 2021-08-02 at 10 40 40
Screenshot 2021-08-02 at 10 40 45

v2 merged with v3
Screenshot 2021-08-02 at 10 40 51
Screenshot 2021-08-02 at 10 40 55

You will notice that in the merged, for example, 5 is mapped to 6.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- space from palette v3 - pavlos

</details>

[CX-1705]: https://artsyproduct.atlassian.net/browse/CX-1705

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 78de2a5863ad8c6532050abea93fe9bdfba62ebb
</p>

File: src/palette/Theme.tsx [hunk 1]
```diff
@@ -137,13 +137,27 @@ const figureOutTheme = (theme: keyof typeof THEMES | ThemeType): ThemeType => {
     purple30: THEMES.v3.colors.blue10, // TODO-PALETTE-V3 replace all purple30 with blue10
     purple5: THEMES.v3.colors.blue10, // TODO-PALETTE-V3 replace all purple5 with blue10
   }
-  // TODO-PALETTE-V3 remove the mapping as the last TODO-PALETTE-V3 to be done
+  // TODO-PALETTE-V3 remove the mapping as the last TODO-PALETTE-V3 to be done for color
+
+  // forcing v3 spaces, unless specifically requiring v2, in which case we use `spaceV2`
+  const mergedSpacesV2WithV3OnTop = {
```
Comment (olerichter00) [inline src/palette/Theme.tsx:143]: I like this name 😄

**Review by olerichter00 — APPROVED** [review]
LGTM! 🚀

File: src/palette/Theme.tsx (hunk 1 above)
Comment (pvinis) [reply src/palette/Theme.tsx:143]: 😅 🤷 trying to be very explicit and verbose.

**Pull request merged**